staging/src/k8s.io/apimachinery/pkg/runtime/swagger_doc_generator.go

```diff
@@ -73,7 +73,9 @@ func fmtRawDoc(rawDoc string) string {
 		case strings.HasPrefix(leading, "TODO"): // Ignore one line TODOs
 		case strings.HasPrefix(leading, "+"): // Ignore instructions to the generators
 		default:
-			if strings.HasPrefix(line, " ") || strings.HasPrefix(line, "\t") {
+			if strings.HasPrefix(line, " ") || strings.HasPrefix(line, "\t") ||
+				// Keep the newline for list, so won't break the markdown
+				strings.HasPrefix(line, "* ") || strings.HasPrefix(line, "- ") {
```

Comment:
I noticed a couple of locations have additional new lines that may be unintentional. (The bold \n in the second bullet)

- "matchPolicy defines how the \"rules\" list is used to match incoming requests. Allowed values are \"Exact\" or \"Equivalent\".\n\n- Exact: match a request only if it exactly matches a specified rule. For example, if deployments can be ...
- "matchPolicy defines how the \"rules\" list is used to match incoming requests. Allowed values are \"Exact\" or \"Equivalent\".\n\n- Exact: match a request only if it exactly matches a specified rule.**\n**For example, if deployments can be...

Perhaps bullet points should need the `line = "\n" + line + "\n"` wrapper and only the newline prefix is enough?

Reply:
It works, but the single newline in the bullet point won't break the line and there're some other generated docs with newlines in the list:
- SyncFailed: The controller encountered an error and wasn't able to compute<strong>\n</strong> the number of allowed disruptions. Therefore no disruptions are<strong>\n</strong> allowed and the status of the condition will be False.

Also from the original comment, the "For example" starts a new line with no indent, so I'm not sure if it's part of the list or if it really wants to start a new line but didn't notice that it has no effect, so keep the newline here may have some meanings.

In this case I think the `line = "\n" + line + "\n"` is the same as `line = "\n" + line + " "` (from markdown list formatting view), so I combined them into one wrapper.

Reply:
Ah I see what you mean. If it's rendered in markdown, then there is no difference between the two. I think stylistically omitting the newline would look better for clients that take the raw text without rendering markdown though. Will leave this as a nit

Reply:
> I noticed a couple of locations have additional new lines that may be unintentional. (The bold \n in the second bullet)

I'm wondering if any of these are fallout from the go1.19 doc reformatting (xref https://github.com/kubernetes/kubernetes/pull/111254#issuecomment-1195794968):

> Sweeping what 1.19 gofmt "autocorrected" in our doc strings, there's a fair number of places we had weird indents that gofmt is turning into new paragraphs, etc. I think this PR should go ahead and merge so we can get go1.19 signal, but a followup to sweep the doc auto format changes in this PR and fix up ones that look wrong would be good.

@dims, did an issue ever get opened for that?

Reply:
I think we first want to make sure our API doc is compatible with go's new support for lists in godoc, and then try to make this follow similar rules (or at least a subset of them):
* https://tip.golang.org/doc/go1.19#go-doc
* https://tip.golang.org/doc/comment#lists

Reply:
I found that some lists still can't be reformatted correctly by go1.19, especially the doc for fields in structs, my original idea was to modify these docs to follow the rule but seems they're common mistakes in many places, so I decided to autocorrect them in the generated swagger doc.
@liggitt will we have an issue to modify those docs?

Reply:
I don't think an issue got opened for https://github.com/kubernetes/kubernetes/pull/111254#issuecomment-1195794968, no

Reply:
Thanks for confirm, if it’s all done with our API doc, I’d like to get this move forward since the rebase is quite annoying

Reply:
A couple things I noticed from the docs that we're definitely missing at the moment:

> Gofmt reformats bullet lists to use a dash as the bullet marker, two spaces of indentation before the dash, and four spaces of indentation for continuation lines.

I see some examples of misindents and bullet points using `*` instead of the standardized `-`

> Gofmt preserves but does not require a blank line between a list and the preceding paragraph. It inserts a blank line between a list and the following paragraph or heading.

This is missing in certain places that we're using `*` to represent bullet points

I think this PR will still be required after fixing our API docs, but maybe the diff will be easier to compare with the API docs fixed beforehand.

Reply:
Since it's a long-standing issue, it's hard to justify it as a high-priority / release-blocking bug. I'd be inclined to merge it for 1.26. A fixup / follow-up for https://github.com/kubernetes/kubernetes/pull/111254#issuecomment-1195794968 would be reasonable to do for 1.25, I think